cfgov/unprocessed/js/modules/ExternalSite.js

```diff
@@ -0,0 +1,82 @@
+/* ==========================================================================
+   External Site Initialization
+   Used on at least `/external-site/`.
+   ========================================================================== */
+
+'use strict';
+
+var atomicCheckers = require( '../modules/util/atomic-checkers' );
+
+/**
+ * ExternalSite
+ * @class
+ *
+ * @classdesc Initializes a new ExternalSite instance.
+ *
+ * @param {HTMLElement} element DOM Element.
+ */
+function ExternalSite( element ) {
+
+  var BASE_CLASS = 'external-site_container';
+  var TOTAL_DURATION = 5;
+  var INTERVAL = 1000;
+
+  var _dom =
+    atomicCheckers.validateDomElement( element, BASE_CLASS, 'ExternalSite' );
```

Comment:
It checks if the element passed to it contains the searched for class OR a child of the passed element contains the searched for class. Essentially you can pass an atomic component in or a container for the atomic component. And yes if it doesn't exist at all it throws a (hopefully) useful error instead of `classList not found on undefined`.